### PR TITLE
arduino-ide: sls for 2.1.x and higher

### PR DESCRIPTION
**Pull request by teclab-ifcfraiburgo**
Created an sls for [Arduino IDE](https://www.arduino.cc/en/software). I've successfully tested on Windows 10.

### CODE REVIEW

File: arduino-ide.sls [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  {% for version, winversion in versions.items() %}
+  '{{ winversion }}':
+    full_name: 'Arduino IDE'
+    installer: 'https://github.com/arduino/arduino-ide/releases/download/{{ version }}/arduino-ide_{{ version }}_Windows_64bit.msi'
+    install_flags: '/qn /norestart'
+    uninstaller: 'https://github.com/arduino/arduino-ide/releases/download/{{ version }}/arduino-ide_{{ version }}_Windows_64bit.msi'
+    uninstall_flags: 'ALLUSERS=1 /qn /norestart'
+    msiexec: True
+    locale: en_US
+    reboot: False
```
Comment (dafyddj) [inline arduino-ide.sls:21]: These are non-functional so only serve to clutter the file.
```suggestion
```

File: arduino-ide.sls [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  }
+%}
+
+arduino-ide:
+  {% for version, winversion in versions.items() %}
+  '{{ winversion }}':
+    full_name: 'Arduino IDE'
+    installer: 'https://github.com/arduino/arduino-ide/releases/download/{{ version }}/arduino-ide_{{ version }}_Windows_64bit.msi'
+    install_flags: '/qn /norestart'
+    uninstaller: 'https://github.com/arduino/arduino-ide/releases/download/{{ version }}/arduino-ide_{{ version }}_Windows_64bit.msi'
```
Comment (dafyddj) [inline arduino-ide.sls:17]: Where the uninstaller is the same as the installer it's not necessary to include the uninstaller field.
```suggestion
```

File: arduino-ide.sls [hunk 3]
```diff
@@ -0,0 +1,22 @@
+{% set versions = {
+  '2.3.2': '2.3.2.0',
+  '2.3.1': '2.3.1.0',
+  '2.3.0': '2.3.0.0',
+  '2.2.1': '2.2.1.0',
+  '2.2.0': '2.2.0.0',
+  '2.1.1': '2.1.1.0'
+  }
+%}
```
Comment (dafyddj) [inline arduino-ide.sls:9]: Please look at `salt-minion.sls` or `firefox.sls` where we are settling on using inline yaml for version lists.
e.g. `{% load_yaml as versions -%}` and it seems to me that `winversion` is redundant; at least at the moment you could just append `.0` in the yaml below.

**Review by dafyddj — COMMENTED** [review]
Thanks for your contribution. If I may offer the following suggested improvements.

File: arduino-ide.sls (hunk 3 above)
Comment (teclab-ifcfraiburgo) [reply arduino-ide.sls]: Thank you for your review. The new commit should apply all your suggestions.

**Review by dafyddj — APPROVED** [review]

**Pull request merged**